pkg/oc/cli/cmd/tag.go

```diff
@@ -180,7 +179,14 @@ func (o *TagOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []s
 
 		sourceKind := o.sourceKind
 		if len(sourceKind) > 0 {
-			sourceKind = determineSourceKind(f, sourceKind)
+			var err error
+			sourceKind, err = determineSourceKind(f, sourceKind)
+			if err != nil {
+				return fmt.Errorf("failed to determine source kind: %v", err)
+			}
+			if len(sourceKind) == 0 {
```

Comment:
You don't need this check, this is being checked outside of this condition, already. All in all, the current changes are only cosmetic, I'm not sure we need it. But thanks for the help :)

Reply:
@soltysh I am sure that it is better than nothing. Even if you support N-1
compatibility, some users accidentally (not accidentally, someone is
lazy to keep updating the cli and server both) may use the latest
command for old server.

Since this issue causes unexpected behavior with the option, it would be
great if you could accept it.